pkg/storegateway/indexheader/binary_reader.go

```diff
@@ -533,84 +534,79 @@ func newFileBinaryReader(path string, postingOffsetsInMemSampling int, cfg Binar
 		return nil, errors.Wrap(err, "read symbols")
 	}
 
-	var lastKey []string
+	var lastLbl labels.Label
+	var lastSet bool
```

Comment:
super-nit: `lastSet := false` (it's just more explicit)

Reply:
This was an internal fight for me, being more explicit vs being consistent with the var declaration above.

Changed in https://github.com/grafana/mimir/pull/3393/commits/30b553389382011be9e2e33f1f9bf7efa2d6774a